src/update/process.rs

```diff
@@ -170,6 +175,10 @@ pub fn process_update(args: UpdateArgs) -> Result<()> {
             style("Authority signature:").bold(),
             authority_signature
         ));
+
+        // Update the cache with the new authority
+        cache.program.update_authority = new_authority_pubkey.to_string();
```

Comment:
I guess this is the reason that the cache is required.

Reply:
It might be better to not store the authority in the cache at all and just make the extra network call to get the current candy state and check the update authority there instead.

Reply:
Yeah, maybe it is a better option. It should ok to do an extra call for that.